test_indirect.cpp

```diff
@@ -12,6 +12,70 @@ TEST_CASE("Defeault construction for indirect", "[constructor.default]")
     REQUIRE(a.operator->() == nullptr); 
 }
 
+TEST_CASE("Copy construction for indirect of a primitive type", "[constructor.copy.primitive]")
+{
+    GIVEN("A value initialised indirect value")
+    {
+        constexpr int a_value = 5;
+        indirect<int> a{ new int(a_value) };
+        REQUIRE(*a == a_value);
+
+        WHEN("Taking a copy of the value initalised value")
+        {
+            indirect<int> copy_of_a{ a };
+            THEN("The copy is a deep copy of the orginal value")
+            {
+                REQUIRE(*copy_of_a == a_value); 
+                REQUIRE(a.operator->() != nullptr); 
+                REQUIRE(copy_of_a.operator->() != nullptr); 
+                REQUIRE(a.operator->() != copy_of_a.operator->()); 
+            }
+        }
+    }
+}
+
+TEST_CASE("Copy assignment for indirect of a primative type", "[assignment.copy.primitive]")
+{
+    GIVEN("A value initialised indirect value")
```

Comment:
I would hyphenate ‘value-initialised’ but am happy without if you feel strongly.

Reply:
No, I agree, I just didn't think about it.  Corrected all instance of it.